### PR TITLE
apps: Obtain and send info about running Apps

### PR DESCRIPTION
**Pull request by mike-sul**
Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

File: src/docker/dockerclient.cc [hunk 1]
```diff
@@ -57,6 +57,82 @@ std::tuple<bool, std::string> DockerClient::getContainerState(const Json::Value&
...
+
+int DockerClient::getContainerExitCode(const std::string& id) {
+  const std::string cmd{"http://localhost/containers/" + id + "/json"};
+  auto resp = http_client_->get(cmd, HttpInterface::kNoLimit);
+  if (!resp.isOk()) {
+    throw std::runtime_error("Request to dockerd has failed: " + cmd);
+  }
+  const auto container_info{resp.getJson()};
+  return container_info["State"]["ExitCode"].asInt();
+}
```
Comment (doanac) [inline src/docker/dockerclient.cc:78]: i know its a draft, but it would be nice if we could make that docker engine api call once and get both values.

Comment (mike-sul) [reply src/docker/dockerclient.cc:78]: We don't both, either health or exit code is requested.

**Comment by mike-sul** [comment]
I wouldn't like to invest more into the api part until someone explicitly ask for it (interface methods to fetch Apps state and send it to DG).

File: src/composeappmanager.cc [hunk 2]
```diff
@@ -530,6 +530,46 @@ std::string ComposeAppManager::getRunningAppsInfoForReport() const {
...
+          if (service["health"] == "unhealthy") {
+            // if it's unhealthy then an overall App is considered unhealthy
+            (*ii)["state"] = "unhealthy";
+            break;
+          }
+        }
+
+        if (!(*ii).isMember("uri")) {
+          // App's URI is stored in the `state` in the case of a regular Compose App engine
+          (*ii)["uri"] = "";  // TODO: figure out an app's URI
```
Comment (doanac) [inline src/composeappmanager.cc:560]: are you still trying to find a way to do this or are you planning to leave it as "TODO" for this PR and do later?

Comment (mike-sul) [reply src/composeappmanager.cc:560]: The later.

Comment (mike-sul) [reply src/composeappmanager.cc:560]: It actually works for `compose apps` case not `restorable one`, we store compose app "state" in a file which contains an app uri.

**Review by doanac — APPROVED** [review]
Looks good.

**Pull request merged**